moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#
+
+import json
+import os
+import time
+from moduleframework import common
+from moduleframework.common import CommonFunctions
+from moduleframework.mtfexceptions import ContainerExc, ConfigExc
+
+
+class OpenShiftHelper(CommonFunctions):
+    """
+    Basic Helper class for OpenShift container module type
+
+    :avocado: disable
+    """
+
+    def __init__(self):
+        """
+        set basic object variables
+        """
+        super(OpenShiftHelper, self).__init__()
+        self.name = None
+        self.docker_id = None
+        self.icontainer = self.get_url()
+        if not self.icontainer:
+            raise ConfigExc("No container image specified in the configuration file or environment variable.")
+        if "docker=" in self.icontainer:
+            self.container_name = self.icontainer[7:]
+        else:
+            # untrusted source
+            self.container_name = self.icontainer
+        # application name is taken from docker.io/modularitycontainer/memcached
+        self.app_name = self.container_name.split('/')[-1]
+        self.app_ip = None
+        common.print_debug(self.icontainer, self.app_name)
+
+    def get_container_url(self):
+        """
+        It returns actual URL link string to container, It is same as URL
+
+        :return: str
+        """
+        return self.icontainer
+
+    def get_docker_instance_name(self):
+        """
+        Return docker instance name what will be used inside docker as docker image name
+        :return: str
+        """
+        return self.container_name
+
+    def _app_exists(self):
+        """
+        It checks if an application already exists in OpenShift environment
+        :return: True, application exists
+                 False, application does not exist
+        """
+        oc_status = self.runHost("oc status", ignore_status=True)
+        if 'dc/%s' % self.app_name in oc_status.stdout:
+            common.print_info("Application already exists.")
+            return True
+        oc_services = self.runHost("oc get services -o json")
+        json_svc = self._convert_string_to_json(oc_services.stdout)
+        if not json_svc["items"]:
```

Comment:
as @TomasTomecek mentioned, probably you need to add there some "__service__" specific calls. I don't know it but probably there will be something similar to `docker container inspect CONTAINER_ID`  what will show state just for one service instead of list all services and try to get some data